### DISCUSSION TITLE
Weekly #8

### DISCUSSION THREAD

**Discussion by Ninetilt** [Ninetilt/Project-TDC#17] [General]
# **Übersicht**
Diese Woche ist bei TDC nicht sonderlich viel passiert. Dennoch wollen wir euch kurz über unsere Pipeline berichten.

## **Pipeline**
Bei dem Pipeline-Tool haben wir uns für GitHub-Actions entschieden. Da wir generell schon GitHub für unser Repository und unseren Blog verwenden, war diese Entscheidung schnell und einfach getroffen.

Die Datei mit den Regeln für die CI/CD-Pipeline findet ihr [hier ](https://github.com/Ninetilt/Project-TDC/blob/main/.github/workflows/TDC_Build_Pipeline.yaml)als yaml-Datei.

Den Hauptpunkt, den wir hier hervorheben möchten ist, dass wir uns dazu entschieden haben bei jedem workflow run buildWindows und buildAndroid ausgeführt wird. Das bedeutet, dass bei jeder Änderung GitHub über eine VM für Windows und Android die Änderungen entsprechend der Pipeline-Vorgaben überprüfen und testen kann.

Sonst ist die Pipeline noch sehr basic gehalten, was Vorgaben für den Code angeht.

Hier wollen wir euch noch ein Paar Screenshots zeigen, wie GitHub-Actions in Aktion aussieht:

// Übersicht aller Worklows und deren Status
![ghactions](https://github.com/user-attachments/assets/b69ec602-9b14-4877-b9e2-4489ebbadc32)

// Build in Porcess
![buildprocess](https://github.com/user-attachments/assets/a04b1d63-cc94-4725-a9be-ad83b7af0deb)

// Build complete
![builddone](https://github.com/user-attachments/assets/fd32cfc9-b3c7-4ef5-8a47-de50ab11044b)

**Comment by Kartoffeleintopf** [comment]
Das TripAhead-Team nutzt ebenfalls GitHub Actions, ist auch wie Ihr sagt die naheliegende Wahl.
> Euer TripAhead-Team (CN)

**Comment by j1tbk** [comment]
Danke für die tolle bildliche Veranschaulichung und passende Erklärung. Die Workflows sind sinnvoll strukturiert und aufgeteilt.  Swipe right und super like.
Euer Team Health
